libpod/container_internal.go

```diff
@@ -1816,18 +1816,22 @@ func (c *Container) cleanupStorage() error {
 	}
 
 	var cleanupErr error
+	reportErrorf := func(msg string, args ...any) {
+		err := fmt.Errorf(msg, args...) // Always use fmt.Errorf instead of just logrus.Errorf(…) because the format string probably contains %w
+		if cleanupErr == nil {
+			cleanupErr = err
+		} else {
+			logrus.Errorf("%s", err.Error())
+		}
+	}
```

Comment:
This is nice but maybe to avoid duplicating this function in wto palces we could export this to another function which just accepts cleanupErr as arg and then returns this anonymous function.

Reply:
I’m not familiar with the codebase and I’m fine with making whatever adjustments are requested. I’ll just push back a bit, once :)

---

My general impression is that this, ~silently proceeding on failures, is a fairly unusual error handling model, and I’m not _too_ enthusiastic about codifying this approach and making it look like an approach to be followed elsewhere. In some ideal end state (which we could never test properly), I imagine we would track which parts of the cleanup succeeded and which are still outstanding, so that the user can re-start the cleanup and get the other parts done.

(Also, so far there are only two functions that use it. My personal, not at all evidence-based, rule of thumb is to consolidate at 3 instances.)